grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -514,7 +514,7 @@ class CycleCountService {
         cycleCountItem.properties = command.properties
         cycleCountItem.countIndex = command.recount ? 1 : 0
         cycleCountItem.status = command.recount ? CycleCountItemStatus.INVESTIGATING : CycleCountItemStatus.COUNTING
-        cycleCountItem.dateCounted = new Date()
+        cycleCountItem.dateCounted = command.dateCounted
```

Comment:
can be there a situation when the command.dateCounted will be null? Don't we need a fallback?